api/db/migrations/20210908085902_add-method-column-in-assessments.js

```diff
@@ -0,0 +1,31 @@
+const TABLE_NAME = 'assessments';
+const COLUMN_NAME = 'method';
+
+exports.up = async function(knex) {
+  await knex.schema.table(TABLE_NAME, (table) => {
+    table.text(COLUMN_NAME);
+  });
+
+  const nbRows = (await knex.raw('select reltuples from pg_class where relname=\'assessments\'')).rows.shift().reltuples;
+  const maxRowCountForUpdate = 650000;
+  const canFillColumn = nbRows < maxRowCountForUpdate;
+
+  if (canFillColumn) {
+    await knex.raw(`UPDATE "assessments" SET "method" =
+      CASE
+        when "type" = 'COMPETENCE_EVALUATION' then 'SMART_RANDOM'
+        when "type" = 'CAMPAIGN' then 'SMART_RANDOM'
+        when "type" = 'PLACEMENT' then 'SMART_RANDOM'
+        when "type" = 'CERTIFICATION' then 'CERTIFICATION_DETERMINED'
+        when "type" = 'DEMO' then 'COURSE_DETERMINED'
+        when "type" = 'PREVIEW' then 'CHOSEN'
+      END;
```

Comment:
Il est possible de faire un petit peu plus léger :
```suggestion
      CASE "type"
        WHEN 'COMPETENCE_EVALUATION' THEN 'SMART_RANDOM'
        WHEN 'CAMPAIGN' THEN 'SMART_RANDOM'
        WHEN 'PLACEMENT' THEN 'SMART_RANDOM'
        WHEN 'CERTIFICATION' THEN 'CERTIFICATION_DETERMINED'
        WHEN 'DEMO' THEN 'COURSE_DETERMINED'
        WHEN 'PREVIEW' THEN 'CHOSEN'
      END;
```